src/app/download/download.component.ts

```diff
@@ -0,0 +1,35 @@
+import {Component} from '@angular/core';
+import {AdAuthoringService} from '../ad-authoring/ad-authoring.service';
+import {Observable} from 'rxjs';
+import {map} from 'rxjs/operators';
+
+@Component({
+  selector: 'app-download',
+  templateUrl: './download.component.html',
+  styleUrls: ['./download.component.scss'],
+})
+export class DownloadComponent {
+  fileUrl;
+  ampHtmlObs: Observable<string>;
+
+  constructor(private service: AdAuthoringService) {
+    this.ampHtmlObs = service
+      .getAdAuthorings()
+      .pipe(map(state => state.AMPHTMLstring));
+  }
+
+  downloadFile() {
+    // TODO: create a service that generates the AMP ad html instead of this dummy data
+    const data = 'amp story ad html goes here';
+    const a = document.createElement('a');
```

Comment:
nit: Move this down below the 'const url' assignment. In general method variables should be defined as close to where their first needed as possible.